todo_app/templates/index.html

```diff
@@ -12,7 +12,33 @@ <h1 class="display-4">To-Do App</h1>
       <h2>Items</h2>
       <ul class="list-group mb-4">
         <!-- List the todo items here -->
+        <table>
+          <tr>
+              {%for col in data[0].keys()%}
+                <th>{{ col }}</th>
+              {%endfor%}
+          </tr>
+          {%for i in range(0, len)%}
+            <tr>
+              {%for col in data[i].values()%}
+                <td>{{ col }}</td>
+              {%endfor%}
+            </tr>
+          {%endfor%}
```

Comment:
Again very minor, but rather than looping over the length here we could just directly loop over the elements:
```suggestion
          {%for entry in data%}
            <tr>
              {%for col in entry.values()%}
                <td>{{ col }}</td>
              {%endfor%}
            </tr>
          {%endfor%}
```
This would just let us skip passing through the len, and arguably make this a bit more readable